core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -151,19 +150,22 @@ object Executor {
     arguments: Map[String, InputValue],
     variableDefinitions: List[VariableDefinition],
     variableValues: Map[String, InputValue]
-  ): Map[String, InputValue] =
-    arguments.map {
-      case (k, v) =>
-        k -> (v match {
-          case InputValue.VariableValue(name) =>
-            lazy val defaultInputValue = (for {
-              definition <- variableDefinitions.find(_.name == name)
-              inputValue <- definition.defaultValue
-            } yield inputValue) getOrElse v
-            variableValues.getOrElse(name, defaultInputValue)
-          case value => value
-        })
-    }
+  ): Map[String, InputValue] = {
+    def resolveVariable(value: InputValue): InputValue =
+      value match {
+        case InputValue.ListValue(values) => InputValue.ListValue(values.map(resolveVariable))
+        case InputValue.ObjectValue(fields) =>
+          InputValue.ObjectValue(fields.map({ case (k, v) => k -> resolveVariable(v) }))
+        case InputValue.VariableValue(name) =>
+          lazy val defaultInputValue = (for {
+            definition <- variableDefinitions.find(_.name == name)
+            inputValue <- definition.defaultValue
+          } yield inputValue) getOrElse value
+          variableValues.getOrElse(name, defaultInputValue)
+        case value: Value => value
+      }
+    arguments.map({ case (k, v) => k -> resolveVariable(v) })
```

Comment:
This is nitpicking but why the parenthesis around the braces? You find this more readable? I usually keep only the braces.

Reply:
Recently I grew a strong reluctance against whitespace-based function application. It feels to irregular and prone to issues (e.g. the order of execution) and so I try to avoid it. 
It's even worse for such an application of functions expecting `PartialFunction` as an argument. I haven't yet seen a junior developer who properly understands that `map { case xx => ..|}`  is, in fact, a partial function literal passed to a `map`. So I prefer to make it explicit and have the regular syntax across the codebase. 

But please, feel free to adjust to your style preference, I wouldn't mind at all!

Reply:
Interesting point. I'm so used to `map { }` that I haven't considered that. I don't mind keeping it that way.